cosmos_predict2/models/text2image_dit.py

```
@@ -1180,9 +1259,13 @@ def __init__(
                     mlp_ratio=mlp_ratio,
                     use_adaln_lora=use_adaln_lora,
                     adaln_lora_dim=adaln_lora_dim,
-                    backend=atten_backend,
+                    self_attention_backend=atten_backend
+                    if natten_parameters is None or natten_parameters[i] is None
+                    else "natten",
```

It might be confusing if the `atten_backend` specified in the config is not `"natten"` and the actual operation is `"natten"`.
Why don't we just let it fail when there exists `natten_parameters` and the atten_backend is not `"natten"` with an error message?

---

This does the opposite. If `natten_parameters` are passed it will use it as specified, otherwise it will respect `atten_backend`.

The only alternative I can think of is making `atten_backend` a list too, which just seems worse.

We can't really control the fact that we want to be able to 1. keep cross-attention layers intact, 2. allow keeping full self attention in the same model that has natten layers, by just passing in `None`s for that layer.